docs/proposals/service_links.md

```diff
@@ -0,0 +1,231 @@
+# Explicit service links
+
+## Abstract
+
+Proposes an API for explicit links to services in the pod spec.
+
+## Motivation
+
+Currently, all pods are automatically injected with docker-link-style
+environment variables for each service in their namespaces.  Making service
+dependencies explicit in the PodSpec has a number of advantages:
+
+1.  Reduce accidental coupling
+2.  Reduce the number of IP tables rules that have to be created by the
+    kube-proxy
+
+## Constraints and Assumptions
+
+1.  The old behavior must remain the default (for now)
+2.  It must be possible to migrate off the default to using explicit links
```

Comment:
I don't want a super-powerful tool to inject fine-grained pieces into single env vars -- it's not needed IMO.  We can add it later if there is a demand for something like that.  I started going down that road and it got super-complex very fast.